consensus/src/pending_votes.rs

```diff
@@ -128,14 +129,37 @@ impl PendingVotes {
         // 3. Let's check if we can create a QC
         //
 
+        let len = self.li_digest_to_votes.len() + 1;
         // obtain the ledger info with signatures associated to the vote's ledger info
-        let li_with_sig = self.li_digest_to_votes.entry(li_digest).or_insert_with(|| {
-            // if the ledger info with signatures doesn't exist yet, create it
-            LedgerInfoWithPartialSignatures::new(
-                vote.ledger_info().clone(),
-                PartialSignatures::empty(),
-            )
-        });
+        let (hash_index, li_with_sig) =
+            self.li_digest_to_votes.entry(li_digest).or_insert_with(|| {
+                // if the ledger info with signatures doesn't exist yet, create it
+                (
+                    len,
+                    LedgerInfoWithPartialSignatures::new(
+                        vote.ledger_info().clone(),
+                        PartialSignatures::empty(),
+                    ),
+                )
+            });
+
+        let validator_voting_power = validator_verifier
+            .get_voting_power(&vote.author())
+            .unwrap_or(0);
+        if validator_voting_power == 0 {
+            warn!("Received vote with no voting power, from {}", vote.author());
+        }
+        let cur_epoch_round = vote.vote_data().proposed().epoch()
+            * counters::EPOCH_ROUND_MULTIPLIER
```

Comment:
This is confusing, why do we need to have this multiplier? Can we just not emit the epoch and round separately?

Reply:
how do I do max then?

Reply:
how would you do something like this:
http://o11y.aptosdev.com/grafana/goto/UyEqWFAVz?orgId=1

Reply:
I see. I am not sure if there is any easier way to do it in Grafana. I think the current way is fine.

Reply:
oh, I am stupid, I can do the multiplication in the Grafana :D let me just have two counters, so that multiplier is not hardcoded

best here would be to have height, but that is not available in vote messages.